### DISCUSSION TITLE
Please Help? In Cascader component, on opening with data={My data} it picks submenu from oldData.

### DISCUSSION THREAD

**Discussion by harshRishi** [rsuite/rsuite#3276] [Q&A] [answered]
In the Cascader component, on opening with data={My data}, it picks the submenu from old data.
So when I change the data and open the cascade again it opens the first column correctly with new data but it picks the second column with old data until I select any option from the first column.

```
<Cascader
     cleanable={false}
     style={{ width: 179 }}
     menuWidth={179}

     size="lg"
     placeholder="Select room"

    value={selectedSpace}
    data={roomTreeData}
    disabledItemValues={roomDisableArray}
    onChange={handleSelectedRoom}
/>
```

here, roomTreeData is changing with different selections  and handleSelectedRoom functions simply setting up the selectedSpace

```
const handleSelectedRoom = (value) => {
    if (value) {
      setSelectedSpace(value);
    }
]

```
Now I can show what is going wrong here, 

1. Initial data and its cascader: https://i.ibb.co/n1SQvth/cap1.png
2. New data and its cascader: https://i.ibb.co/Yk3DFBz/Cap2.png
3. Again go to the previous data and its cascader:  https://i.ibb.co/HDRmC4g/cp3.png

we can see in the first image floor - 1 has {room -0, 1, 10....so on}
but when we change the data to new data then in image 2 we can observe building -1 has {tower - 1 and tower - 2}
but again when we change the data to initial data then it gets confusing, as we already know that floor - 1 is not even selected in image 3 but still it renders the submenu with Tower - 1 and Tower - 2 which was the submenu of the new Data.

Please help guys!
HARSH RISHI.

**Comment by SevenOutman** [comment]
Hi @harshRishi, did you try clearing `value` when the data changes?

**Reply by harshRishi** [reply]
Yes, I've used a useEffect hook to clear the value (ie. selectedSpace).
```
const [selectedSpace, setSelectedSpace] = useState(""); // initial state

 useEffect(() => {
    setSelectedSpace("");
 }, [roomTreeData]);
```
@SevenOutman

**Reply by SevenOutman** [reply]
Could you provide a CodeSandbox demo? You may use this template https://codesandbox.io/s/rsuite-template-jdydg7

**Reply by harshRishi** [reply, marked as answer]
I can't provide the code and data from my company, but I have somehow achieved the desired behavior.
Thanks for your assistance @SevenOutman
